src/runtime/recipe/handle-connection.ts

```diff
@@ -78,31 +78,34 @@ export class HandleConnection {
     return 0;
   }
 
-  get recipe() { return this._recipe; }
-  get name() { return this._name; } // Parameter name?
-  getQualifiedName() { return `${this.particle.name}::${this.name}`; }
-  get tags() { return this._tags; }
+  get recipe(): Recipe { return this._recipe; }
+  get name(): string { return this._name; } // Parameter name?
+  getQualifiedName(): string { return `${this.particle.name}::${this.name}`; }
+  get tags(): string[] { return this._tags; }
+
   get type() {
     if (this.resolvedType) {
       return this.resolvedType;
     }
     const spec = this.spec;
     return spec ? spec.type : null;
   }
-  get direction() { // in/out/inout/host/consume/provide
+
+  get direction(): Direction|undefined { // in/out/inout/host/consume/provide
```

Comment:
I think there should be a space surrounding the `|`.

Reply:
We should also include `null` as a return. 
Is there an alias in the codebase or ts lang that unions `undefined` and `null`?

Reply:
Not sure, I'd have to check.  Turning on `strictNullChecks` is going to be a lot of work...